oldsdb.js

```diff
@@ -25,48 +25,94 @@ const db = mysql.createPool({
     timezone: 'Z'
 });
 
-const dbGetRecord = (sql, session) => {
-console.log(sql, session)
+const dbRunSQL = (sql, subst) => {
+    return new Promise(((resolve, reject) => {
+        db.query(sql,
+                subst,
+                (error, result) => {
+                let op = sql.split(" ")[0].toUpperCase();
+                    if(error) {
+                        console.log("DB ${op} ERROR: ", error.sqlMessage);
+                        console.log("Data    : ", subst);
+                        reject(`${op} failed: ${ error.sqlMessage}`);
+                        }
+
+                    switch(op){
+                        case "INSERT":
+                            res = result ? result.insertId : null;
+                            break;
+                        case "UPDATE":
+                            res = result ? result.affectedRows : null;
+                            if(res===0){
+                                reject(`${op} failed: ${result.message} :: [${subst}]`);
+                            }
+                            break;
+                        case "DELETE":
+                            res = "OK";
+                            break;
+                        default:
+                            res = result ? result : [];
+                    }
+                    resolve(res)
+                })
+        }))
+    }
+
+const getColumns = () => {
+    let sql = `
+        SELECT TABLE_NAME as tbl,
+            COLUMN_NAME as col,
+            IF(IS_NULLABLE='Yes',1,0) as "null",
+            IF(COLUMN_KEY IN ('PRI'),1,0) as "key",
+            IF(EXTRA = 'auto_increment',1,0) as auto
+        FROM information_schema.COLUMNS WHERE Table_schema = 'oldsdb';
+        `;
     return new Promise((resolve => {
-        db.query(sql, session, (error, result) => {
-                if(error) console.log("ERROR selecting top game scores ", error, session);
+        db.query(sql, (error, result) => {
+                if(error) console.log("ERROR : ", error);
                 resolve(result ? result : [])
             })
     }))
-}
 
-
-
-const newRecords = async (req, res) => {
 };
 
-const updateRecord = async (req, res) => {
-};
-
-const getRecords = async (req, res) => {
-
-    let table = req.params.tbl.toLowerCase();
-    let table_name = tables[table];
-    if ( !table_name ) {
-        throw new ServerError('---');
+const getTables = async () => {
+    let tables = loadSchema('oldsdb/tables.json')
+    let columns = await getColumns()
+    for (let n in columns){
```

Comment:
use: for(column of columns) { ... }
then you can shorten the code: columns[n] -> column